src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            }
+
+            //delete the value that we created for backup purpose
+            _registryManager.DeleteValue(targetHive, subKeyName, nameofTheBackupValue);
+        }
+
+        private void RevertRegistrySettingsForCurrentUser()
+        {
+            var hive = RegistryHive.CurrentUser;
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.ScreenSaver, RegistryConstants.ValueNames.ScreenSaver);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.ScreenSaverDomainPolicy, RegistryConstants.ValueNames.ScreenSaver);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.StartupProcess, RegistryConstants.ValueNames.StartupProcess);
+        }
+
+        private void RevertRegistrySettingsForDifferentUser(string securityId)
+        {
+            var hive = RegistryHive.Users;
+            RevertOriginalValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaver}", RegistryConstants.ValueNames.ScreenSaver);
+            RevertOriginalValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaverDomainPolicy}", RegistryConstants.ValueNames.ScreenSaver);
+            RevertOriginalValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.StartupProcess}", RegistryConstants.ValueNames.StartupProcess);
+        }
+
+        private void SetValue(RegistryHive targetHive, string subKeyName, string name, string value)
+        {
+            //take backup if it exists
+            string origValue = _registryManager.GetValue(targetHive, subKeyName, name);
+            if (!string.IsNullOrEmpty(origValue))
+            {
+                var nameForTheBackupValue = GetBackupValueName(name);
+                _registryManager.SetValue(targetHive, subKeyName, nameForTheBackupValue, origValue);
+            }
+
+            _registryManager.SetValue(targetHive, subKeyName, name, value);
+        }
+
+        private string GetBackupValueName(string valueName)
+        {
+            return string.Concat(RegistryConstants.BackupKeyPrefix, valueName);
+        }
+
+        private void ShowAutoLogonWarningIfAlreadyEnabled(string domainName, string userName)
+        {
+            //we cant use store here as store is specific to the agent root and if there is some other on the agent, we dont have access to it
+            //we need to rely on the registry only
+            FetchAutoLogonUserDetails(out string autoLogonUserName, out string autoLogonUserDomainName);
+            
+            if (autoLogonUserName != null
+                    && autoLogonUserDomainName != null
+                    && !domainName.Equals(autoLogonUserDomainName, StringComparison.CurrentCultureIgnoreCase)
+                    && !userName.Equals(autoLogonUserName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                var terminal = HostContext.GetService<ITerminal>();
+                terminal.WriteLine(StringUtil.Loc("AutoLogonAlreadyEnabledWarning", userName));
```

Comment:
as mentioned, it is not a termination point so why to put as error?

---
In reply to: [118599005](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118599005) [](ancestors = 118599005)